File: 2015/01_Not_Quite_Lisp/part_1/main.kt
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 1.1
 */

import java.io.File

fun main(args: Array<String>) {
  var floor = 0 // Start at floor 0

  /*
    '(' = ascend floor
    ')' = descend floor
   */
  File("../input.txt").readLines()[0]
    .forEach { floor = if (it == '(') floor + 1 else floor - 1 }

  println(floor)
}

File: 2015/01_Not_Quite_Lisp/part_1/main.js
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 1.1
 */

let input = null;

try {
  input = require('fs').readFileSync("../input.txt", "utf-8").split("");
} catch (err) {
  console.error(err);
}

if (input !== null) {
  let floor = 0; // Start at floor 0

  while (input.length > 0) {
    /*
      '(' = ascend floor
      ')' = descend floor
     */
    floor = input.shift() == '(' ? floor + 1 : floor - 1;
  }

  console.log(floor);
}
else console.error("Failed to read input.txt");